packages/plugin-core/src/commands/MoveHeader.ts

```diff
@@ -0,0 +1,311 @@
+        if (_.has(a, "children")) {
+          return _.every(a.children as Node[], (aChild, aIndex) => {
+            const bChild = (b.children as Node[])[aIndex];
+            return this.hasIdenticalChildren(aChild, bChild);
+          });
+        }
+        return true;
+      } else {
+        return false;
+      }
+    } else {
+      return false;
+    }
+  };
+
+  async gatherInputs(opts: CommandInput): Promise<CommandOpts | undefined> {
+    const engine = getEngine();
+    const { editor, selection } = VSCodeUtils.getSelection();
+    if (!editor) throw this.noActiveNoteError;
+    if (!selection) throw this.headerNotSelectedError;
+
+    // const text = editor.document.getText();
+    const line = editor.document.lineAt(selection.start.line).text;
+    const maybeNote = VSCodeUtils.getNoteFromDocument(editor.document);
+    if (!maybeNote) {
+      throw this.noActiveNoteError;
+    }
+
+    const proc = this.getProc(engine, maybeNote);
+    const parsedLine = proc.parse(line);
+    let targetHeader: Heading | undefined;
+    visit(parsedLine, [DendronASTTypes.HEADING], (heading: Heading) => {
+      targetHeader = heading;
+      return false;
+    });
+    if (!targetHeader) {
+      throw this.headerNotSelectedError;
+    }
+    const tree = proc.parse(maybeNote.body);
+    let headerFound = false;
+    let foundHeaderIndex: number | undefined;
+    let nextHeaderIndex: number | undefined;
+    visit(tree, (node, index) => {
+      if (nextHeaderIndex) {
+        return;
+      }
+      const depth = node.depth as Heading["depth"];
+      if (!headerFound) {
+        if (node.type === DendronASTTypes.HEADING) {
+          if (
+            depth === targetHeader!.depth &&
+            this.hasIdenticalChildren(node, targetHeader!)
+          ) {
+            headerFound = true;
+            foundHeaderIndex = index;
+            return;
+          }
+        }
+      } else if (node.type === DendronASTTypes.HEADING) {
+        if (foundHeaderIndex) {
+          if (depth <= targetHeader!.depth) nextHeaderIndex = index;
+        }
+      }
+    });
+
+    const nodesToMove = nextHeaderIndex
+      ? (tree.children as any[]).splice(
+          foundHeaderIndex!,
+          nextHeaderIndex! - foundHeaderIndex!
+        )
+      : (tree.children as any[]).splice(foundHeaderIndex!);
+
+    const lookupController = LookupControllerV3.create({
+      nodeType: "note",
+      disableVaultSelection: true,
+    });
+    const lookupProvider = new NoteLookupProvider(this.key, {
+      allowNewNote: false,
+      noHidePickerOnAccept: false,
+    });
+
+    lookupController.show({
+      title: "Select note to move header to",
+      placeholder: "note",
+      provider: lookupProvider,
+      initialValue: opts?.dest ? opts.dest.fname : undefined,
+      nonInteractive: !_.isUndefined(opts?.dest),
+    });
+
+    return new Promise((resolve) => {
+      HistoryService.instance().subscribev2("lookupProvider", {
```

Comment:
as a separate item, we should abstract this away into a utility that all lookup commands can more easily adopt

Reply:
added to my backlog 👍🏼